CMakeLists.txt

```diff
@@ -36,29 +39,31 @@ target_link_libraries(LibEugeneCore
         CONAN_PKG::fmt
         CONAN_PKG::ms-gsl
         CONAN_PKG::outcome
+        CONAN_PKG::catch2
         CONAN_PKG::spdlog
-        CONAN_PKG::catch2)
+        CONAN_PKG::cppbenchmark)
+
 
 if (DEFINED ENV{EUGENE_BUILD_TESTS})
-    macro(eugene_test component path_prefix)
-        add_executable(Test${component} ${path_prefix}/${component}.cpp)
-        target_link_libraries(Test${component} TestMain LibEugeneCore)
-        add_test(NAME Test${component}
-                COMMAND Test${component})
+    macro(eugene_test component path_prefix test_name)
+        add_executable(${component}${test_name} ${path_prefix}/${component}.cpp)
+        target_link_libraries(${component}${test_name} TestMain LibEugeneCore)
+        add_test(NAME ${component}${test_name}
+                COMMAND ${component}${test_name})
         message("   -- Test about '${component}' component added from \"${path_prefix}/${component}.cpp\"")
     endmacro()
 
     message("-- Building core unit tests")
 
     add_library(TestMain src/core/Main.cpp)
-    target_link_libraries(TestMain CONAN_PKG::catch2)
```

Comment:
I think these changes do not belong in this PR. Please move them to a separate one.

Reply:
Ok, my bad